lib/apr/views/commerce/commerce_transaction_slack_view.ex

```diff
@@ -9,129 +9,139 @@ defmodule Apr.Views.CommerceTransactionSlackView do
+      |> append_participant_info(buyer, seller)
+      |> append_transaction_info(event)
 
     %{
-      text:
-        ":alert: <#{stripe_search_link(event["properties"]["order"]["id"])}|#{event["properties"]["failure_code"]}>",
-      attachments: [
+      text: ":alert:",
+      attachments: attachments,
+      unfurl_links: true
+    }
+  end
+
+  defp append_participant_info(attachments, buyer, seller) do
+    attachments ++
+      [
         %{
-          color: "#6E1FFF",
-          author_name: event["properties"]["order"]["code"],
-          author_link: exchange_admin_link(event["properties"]["order"]["id"]),
+          color: "#ED553B",
+          author_name: cleanup_name(buyer["name"]),
+          author_link: exchange_user_orders_link(buyer["_id"]),
           title: seller["name"],
           title_link: exchange_partner_orders_link(seller["_id"]),
-          fields: fields
+          fields:
+            [
+              %{
+                title: "User Since",
+                value: format_datetime_string(buyer["created_at"]),
+                short: true
+              }
+            ] ++ seller_admin(seller)
         }
-      ],
-      unfurl_links: true
-    }
+      ]
+  end
+
+  defp append_order_info(attachments, order) do
+    attachments ++
+      [
+        %{
+          color: "#20639B",
+          author_name: order["code"],
+          author_link: exchange_admin_link(order["id"]),
+          fields:
+            [
+              %{
+                title: "#{order["mode"]} / #{order["fulfillment_type"]}",
+                value:
+                  format_price(
+                    order["items_total_cents"],
+                    order["currency_code"]
+                  ),
+                short: false
+              }
+            ] ++ fulfillment_info(order)
+        }
+      ]
   end
 
-  defp basic_fields(event, buyer) do
+  defp append_transaction_info(attachments, event) do
+    attachments ++
+      [
+        %{
+          color: "#6E1FFF",
+          title: event["properties"]["failure_message"],
+          title_link: stripe_search_link(event["properties"]["order"]["id"]),
+          author_name: event["properties"]["failure_code"],
+          author_link: stripe_search_link(event["properties"]["order"]["id"]),
+          fields:
+            [
+              %{
+                title: "Transaction Type",
+                value: event["properties"]["transaction_type"],
+                short: true
+              }
+            ] ++ stripe_fields(event["properties"]["external_id"], event["properties"]["external_type"])
+        }
+      ]
+  end
+
+  defp fulfillment_info(order = %{"fulfillment_type" => "ship"}) do
+    [
+      %{title: "Shipping Country", value: order["shipping_country"], short: true},
+      %{title: "Shipping Name", value: cleanup_name(order["shipping_name"]), short: true},
+      %{title: "Shipping State", value: order["shipping_region"], short: true}
```

Comment:
I haven't seen a province/region/state for addresses in the UK. Would it make some (otherwise innocent) failed payments from the UK look suspicious if this field were to be `nil`?

Reply:
this is mostly monitored by CRT, if UK addresses don't provide region in general I think CRT would ignore this field for them but will double checking with them. good call!